internal/testutils/jimmtest/e2e_suite.go

```diff
@@ -142,9 +147,65 @@ func (s *WebsocketE2ESuite) SetUpTest(c *gc.C) {
 	s.APIHandler = mux
 	s.HTTP = httptest.NewTLSServer(s.APIHandler)
 
+	s.AddAdminUser(c, "alice@canonical.com")
+
+	cct := names.NewCloudCredentialTag(TestE2ECloudName + "/charlie@canonical.com/cred")
+	cloudCredentials := s.GetExistingClientCredentialsForCloud(c, TestE2ECloudName)
+	s.UpdateCloudCredential(c, cct, cloudCredentials)
+	s.Credential2 = new(dbmodel.CloudCredential)
+	s.Credential2.SetTag(cct)
+	err := s.JIMM.Database.GetCloudCredential(ctx, s.Credential2)
+	c.Assert(err, gc.Equals, nil)
+	mt := s.AddModel(c, names.NewUserTag("charlie@canonical.com"), petname.Generate(2, "-"), names.NewCloudTag(TestE2ECloudName), TestE2ECloudRegionName, cct)
+	s.Model2 = new(dbmodel.Model)
+	s.Model2.SetTag(mt)
```

Comment:
add these for cleanup here

Reply:
This can be even more generic. Add a method `Cleanup(f func())` which adds the function to a slice and iterate over the funcs in the cleanup The benefit of this is that we can more easily drop-in replace gocheck with a regular `*testing.T` which already has the same method with the same signature `func (c *testing.common) Cleanup(f func())`